src/frame/vulkan.rs

```diff
@@ -32,91 +32,98 @@ pub struct Vulkan {
 impl Vulkan {
     pub fn new() -> Result<Self, Box<dyn Error>> {
         let app_name = CString::new("wluma")?;
-        let app_info = vk::ApplicationInfo::builder()
+        let app_info = vk::ApplicationInfo::default()
             .application_name(&app_name)
             .application_version(WLUMA_VERSION)
             .engine_name(&app_name)
             .engine_version(WLUMA_VERSION)
             .api_version(VULKAN_VERSION);
 
         let instance_extensions = &[
-            vk::KhrExternalMemoryCapabilitiesFn::name().as_ptr(),
-            vk::KhrGetPhysicalDeviceProperties2Fn::name().as_ptr(),
+            vk::KHR_EXTERNAL_MEMORY_CAPABILITIES_NAME.as_ptr(),
+            vk::KHR_GET_PHYSICAL_DEVICE_PROPERTIES2_NAME.as_ptr(),
         ];
 
         let entry = Entry::linked();
 
-        let create_info = vk::InstanceCreateInfo::builder()
+        let create_info = vk::InstanceCreateInfo::default()
             .application_info(&app_info)
+            //.enabled_extension_count(instance_extensions.len() as u32)
             .enabled_extension_names(instance_extensions);
 
         let instance = unsafe {
             entry
                 .create_instance(&create_info, None)
-                .map_err(|e| Box::new(e) as Box<dyn Error>)?
+                .map_err(anyhow::Error::msg)?
         };
 
         let physical_devices = unsafe {
             instance
                 .enumerate_physical_devices()
-                .map_err(|e| Box::new(e) as Box<dyn Error>)?
+                .map_err(anyhow::Error::msg)?
         };
         let physical_device = *physical_devices
             .first()
             .ok_or("Unable to find a physical device")?;
 
         let queue_family_index = 0;
-        let queue_info = &[vk::DeviceQueueCreateInfo::builder()
+        let queue_info = &[vk::DeviceQueueCreateInfo::default()
             .queue_family_index(queue_family_index)
-            .queue_priorities(&[1.0])
-            .build()];
+            //p_queue_priorities: [1.0f32].as_ptr(),
```

Comment:
> Do you want to remove these commented parts, here and in a couple of other places, or you are experimenting with them?

Ah! Sorry, I forgot about that, 
I was commenting out some of my code in case if while changing my code it breaks I can come back and revert it.